torch/fx/experimental/symbolic_shapes.py

```diff
@@ -791,29 +891,71 @@ def _make_node_sizes_strides(method, func):
     def sizes_strides_impl(self, sizes, strides):
         op = getattr(sys.modules[__name__], method)
         if SYM_FUNCTION_MODE:
-            r = _handle_sym_dispatch(op, ([wrap_node(s) for s in sizes], [wrap_node(s) for s in strides]), {})
-            assert isinstance(r, SymBool), type(r)
-            return r.node
+            return to_node(
+                self,
+                _handle_sym_dispatch(
+                    op,
+                    ([wrap_node(s) for s in sizes], [wrap_node(s) for s in strides]),
+                    {}
+                )
+            )
         size_exprs = [s.expr for s in sizes]
         stride_exprs = [s.expr for s in strides]
         try:
-            out = func(*size_exprs, *stride_exprs)
+            out = func(size_exprs, stride_exprs)
         except Exception:
-            log.warning(f"failed to eval {method}(*{size_exprs}, *{stride_exprs})")
+            log.warning(f"failed to eval {method}({size_exprs}, {stride_exprs})")
             raise
-        hints = []
+        # bool is never expandable
+
+        size_hints = []
         out_hint = None
-        for s in itertools.chain(sizes, strides):
+        for s in sizes:
             if s.hint is None:
                 break
-            hints.append(s.hint)
+            size_hints.append(s.hint)
         else:
-            out_hint = op(*hints)
-        # bool is never expandable
-        return SymNode(sympy.Eq(out, 1), self.shape_env, bool, out_hint)
+            stride_hints = []
+            for s in strides:
+                if s.hint is None:
+                    break
+                stride_hints.append(s.hint)
+            else:
+                out_hint = op(size_hints, stride_hints)
+
+        # NB: This is the indicator function, not the actual bool!
+        pytype: Type
+        if method.endswith("_indicator"):
```

Comment:
That feels brittle? Don't we know which methods are indicator?

Reply:
I mean, if you like, I can list them out explicitly, but we pick the names for the functions, and I was hoping to just use the indicator convention as an easy way to distinguish which ones directly returned bools versus int